pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -345,27 +345,45 @@ func MakeSegmentStoreHeadlessService(pravegaCluster *api.PravegaCluster) *corev1
 	}
 }
 
-func MakeSegmentStoreExternalServices(pravegaCluster *api.PravegaCluster) []*corev1.Service {
-	var service *corev1.Service
-	var ssPodName string
+func cloneMap(sourceMap map[string]string) (annotationMap map[string]string) {
+	if len(sourceMap) == 0 {
+		return map[string]string{}
+	}
+	annotationMap = make(map[string]string, len(sourceMap)+1)
```

Comment:
why do we need to explicitly specify the map capacity here? The size of the map is managed by GO runtime automatically I think.

Reply:
Since we're cloning the sourceMap, it makes sense to set capacity for the destinationMap, as size is known at the time of creation. Do you see any concerns with doing this?